### DISCUSSION TITLE
Headers for HTTP connector...

### DISCUSSION THREAD

**Discussion by vogler75** [apache/streampipes#1450] [General]
REST interfaces are hopefully somehow protected by a key/token.
Often this token is transmitted by the authentication header (for example a Bearer Token).
It would be good to be able to set additional headers.

**Comment by tenthe** [comment]
Hello @vogler75 ,

That is a good point. I just took a look and saw that we used to offer the option to add an access token, but that was commented out at some point, see [1].
We should turn this discussion into an issue and mark it as a "good first issue" as I think this is a good starting point for new developers.

How should a user provide the configurations?
Currently there are options for a **URL** and a **polling interval**. So we would need an additional option for an **access token**. Are there any other options that should be considered?

[1] https://github.com/apache/streampipes/blob/dev/streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/protocol/stream/HttpStreamProtocol.java

**Reply by bossenti** [reply]
+1 for making a good-first-issue out of it

As an alternative we could also allow to pass `http headers` as key-value pairs so that we are completely generic.
Although, we could till offer certain convenience options, e.g. for Bearer tokens

**Reply by vogler75** [reply]
I would also vote for key value pairs. One or many. Would be the most flexible way.

**Reply by dominikriemer** [reply]
I created an issue for this: #1458
